### PR TITLE
Add Regex Language

### PR DESCRIPTION
**Pull request by thecoolwinter**
Adds support for the regex language. Regex doesn't have a standalone file type so the `extensions` array is left empty.

In action in CodeEdit *(CE doesn't syntax highlight any punctuation pairs but it shows the @ symbol and numbers being parsed)*:
Screenshot 2023-03-15 at 11 43 16 PM

### CODE REVIEW

**Review by lukepistrol — COMMENTED** [review]
Please also add regex to the documentation catalog and readme. Otherwise looks good

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -670,6 +674,14 @@
 				kind = branch;
 			};
 		};
+		6CAB9CA029C2CE0E006B8521 /* XCRemoteSwiftPackageReference "tree-sitter-regex" */ = {
+			isa = XCRemoteSwiftPackageReference;
+			repositoryURL = "https://github.com/thecoolwinter/tree-sitter-regex";
```
Comment (lukepistrol) [inline CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:679]: Please make sure to create a `feature-spm` branch on your fork so everyone knows, that this is not the original repo.

**Comment by lukepistrol** [comment]
I guess the parser does not support highlighting like on `regexr.com`?

Screenshot 2023-03-16 at 11 20 36

**Comment by thecoolwinter** [comment]
> I guess the parser does not support highlighting like on `regexr.com`?
> 
> Screenshot 2023-03-16 at 11 20 36

It does, but it's registering them as captures that CE doesn't yet support:
```
regex [{26, 45}]
  (pattern {26, 45} 
    (term {26, 45} 
      (start_assertion {26, 1} )
      (character_class_escape {27, 2} )
      (one_or_more {29, 1} )
      (anonymous_capturing_group {30, 11} 
        (pattern {31, 9} 
          (term {31, 9} 
            (character_class {31, 5} 
              (identity_escape {32, 2} )
              (class_character {34, 1} ))
            (optional {36, 1} )
            (character_class_escape {37, 2} )
            (one_or_more {39, 1} ))))
      (zero_or_more {41, 1} )
      (pattern_character {42, 1} )
      (character_class_escape {43, 2} )
      (one_or_more {45, 1} )
      (anonymous_capturing_group {46, 11} 
        (pattern {47, 9} 
          (term {47, 9} 
            (character_class {47, 5} 
              (identity_escape {48, 2} )
              (class_character {50, 1} ))
            (optional {52, 1} )
            (character_class_escape {53, 2} )
            (one_or_more {55, 1} ))))
      (zero_or_more {57, 1} )
      (anonymous_capturing_group {58, 11} 
        (pattern {59, 9} 
          (term {59, 9} 
            (identity_escape {59, 2} )
            (character_class_escape {61, 2} )
            (count_quantifier {63, 5} 
              (decimal_digits {64, 1} )
              (decimal_digits {66, 1} )))))
      (one_or_more {69, 1} )
      (end_assertion {70, 1} )))
```

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj (hunk 1 above)
Comment (thecoolwinter) [reply CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj]: Done 👍

**Review by lukepistrol — APPROVED** [review]

**Pull request merged**